ynput/ayon-core
# Unreal: Deadline support with Perforce

## Changelog Description
This PR should add possibility to render Unreal project on Deadline with Perforce integration (and possibly without P4). 

## Additional info
It requires https://github.com/ynput/ayon-version-control/pull/2, working Perforce server with user credentials and P4 workspace created following some P4 depot and stream

## Testing notes:
1. set info about P4
![image](https://github.com/ynput/ayon-core/assets/4457962/48af3c89-b870-4b91-a4cc-89b370e884cd)
2,provide P4 user credentials (in `Project Settings > Site Settings`)
![image](https://github.com/ynput/ayon-core/assets/4457962/e4677c2c-f737-49f0-8ae3-791faf53c7b6)
3. Deadline must have some environment variables set
 - either via OS environments
 - OR via env vars set on Unreal in `Applications` addon
![image](https://github.com/ynput/ayon-core/assets/4457962/51fcc106-aa69-4860-bd05-1d6387e9149f)
Expected env var: (applicable values for DL WORKER, not artist machine)
```
   "UnrealExecutable": {"windows": "C:/Program Files/Epic Games/UE_5.3/Engine/Binaries/Win64/UnrealEditor-Cmd.exe"},
   "UnrealExecutable_5_3": {"windows": "C:/Program Files/Epic Games/UE_5.3/Engine/Binaries/Win64/UnrealEditor-Cmd.exe"},
   "P4PORT":"localhost:1666",
   "P4USER":"admin",
   "P4PASSWD":"pass"
   ```
4. Copy `ayon_core/modules/deadline/repository/custom/plugins/UnrealEngine5` to `DEADLINE_REPO/custom` folder
5. Render for DL uses existing creator and follow existing workflow, only added `Render target`
![image](https://github.com/ynput/ayon-core/assets/4457962/6fec5a4e-59c2-4b7f-8a2c-8e39bdc9cd30)
6. Provide path for render config `ayon+settings://unreal/render_config_path` and make sure that it exists in Unreal project
![image](https://github.com/ynput/ayon-core/assets/4457962/f843e3eb-9a81-4359-a0f9-fca703846d9f)
7. There must be at least 2 instances to be published, `publish_commit` creates instance with information about change list
![image](https://github.com/ynput/ayon-core/assets/4457962/f1eb0b0c-57dc-47e4-893f-794050f44aca)
